server/src/main/java/org/candlepin/hostedtest/HostedTestSubscriptionServiceAdapter.java

```diff
@@ -318,7 +318,7 @@ public ProductInfo updateProduct(String productId, ProductInfo pinfo) {
         // Apply updates...
         // Don't allow changing the ID
 
-        Collection<ProductContentData> pcdata = this.resolveProductContent(pinfo.getProductContent());
+        Collection<ProductContentData> pcdata = this.resolveProductContent(pdata.getProductContent());
```

Comment:
Why is this changed? We want to resolve the product content on the incoming data (pinfo) not the existing product (pdata)

Reply:
This was done to pass `refresh_pools` specs. 
Thanks for pointing it out. This is now reverted. 
The issue was while translating ProductDTO to the Product entity model, the setters were setting collection to empty collection by default. Now updated the Getter/Setter to fulfill this requirement of the Product.

Reply:
I don't think it is a good idea to change our Product model entity to have nullable collections/maps, because it might cause unforeseen problems. I can't think of something specific, but I'm worry of changing this. (e.g. what happens if something sets the collection to null and then hibernate tries to persist the Product?). For me, it's ok to leave this as is just to get the PR merged through for now, but we'd have to create a new card to deal with this later on. I'll leave Chris to re-review/merge if he thinks this is ok temporarily.

@sonalidhome @Ceiu This problem demonstrates why using a model entity as DTO is bad. Model entities have different requirements (non-nullable collections/maps) than DTOs (nullable collections/maps and fields in general). This is something a translation to a DTO (instead of an entity) would solve, but obviously, as Chris said, having multiple translations for the same data in one request is redundant. This is also where the *Info interfaces on the api DTOs where kind of useful as a workaround and we could just pass in the existing DTO. So if we: 
1) can't use the entity model as a DTO because it can't have nullable collections
2) can't translate to new DTOs because that would be redundant
3) can't have the openapi-generated DTOs implement *Info
then what can we do? All the solutions I can think of are hacks:

- We might be able to technically get n3 working by implementing another openapi-generator hack, but in principle I think it is wrong to use the same interfaces on two different APIs. Just because the *info interface is of help in this case as a crutch, it doesn't mean that it is not holding us back in other cases where a field is clearly not needed in one of the 2 APIs (e.g. putting a getter that always returns null on the REST DTO because that field is not needed there, just because the interface is forcing us to implement it).
- We could do n1, but instead of using the model entity (e.g. Product), have a new entity that extends it and overrides the collection/map getters/setters to make collections nullable, but that feels like a worse hack than the other one...